### PR TITLE
Making it possible to access the pluginmanager in the pytest_addoptio…

### LINKED ISSUE #6061: Make it possible to call hooks in pytest_addoption
**Suggestion**

Make it possible to call hooks in pytest_addoption by passing the pluginmanager as an argument. 

I have a  plugin that is used across a number of projects. That plugin has a command line option that is added in pytest_addoption, but the default values for those options will change based on the project/conftest.py that's using them (think: configuration file). While I could use the parser's **extra_info** field to inject the default values, that seems like a hack and not the field's intended use. Instead, it would seem to make more sense to have the conftest.py plugin in each project implement a hook or hooks that return the default values. That way there is a nicely defined contract between the plugin and the conftest.py that depends on it. Unfortunately, the pytest_addoption hook does not make the pluginmanager available as an argument. The plugin can cheat by storing a reference to the pluginmanager in pytest_addhooks, which works like this but feels like a hack:

In the plugin file, myplugin.py:

```python
import hooks

global thepluginmanager

def pytest_addhooks(pluginmanager):
    global thepluginmanager
    pluginmanager.add_hookspecs(hooks)
    thepluginmanager = pluginmanager

def pytest_addoption(parser):
    global thepluginmanager
    parser.addoption("--some-option", help="Some option, defaults to %(default)s",
                     default=thepluginmanager.hook.pytest_myplugin_default_value()[0])
```
In the conftest.py file:

```python
pytest_plugins = ["myplugin"]

def pytest_myplugin_default_value():
    return 1
```
In the hooks.py file:

```python
def pytest_myplugin_default_value():
    """
    Returns the default value for the command line option
    """
```

One possible solution would be to modify the signature to:

```python
def pytest_addoption(parser, pluginmanager)
```
It appears the pluggy can handle this in a backwards compatible fashion, though I could imagine there is a deprecation policy in this project that requires defining a newly named function. Alternatively, add a new hook and call them both:

```python
def pytest_addoption_with_plugins(parser, pluginmanager)
```

**pytest Version: 3.7.4**

### PR DESCRIPTION
**Pull request by joshuastorck**
This resolves #6061.

### CODE REVIEW

File: doc/en/example/markers.rst [hunk 1]
```diff
@@ -364,7 +364,7 @@ specifies via named environments:
     import pytest
 
 
-    def pytest_addoption(parser):
+    def pytest_addoption(parser, pluginmanager):
```
Comment (RonnyPfannschmidt) [inline doc/en/example/markers.rst]: its not  necessary to add the parameters where they arent used

Comment (joshuastorck) [reply doc/en/example/markers.rst]: Just thought it might be confusing for the reader with the change. I can revert.

**Comment by joshuastorck** [comment]
Do I need to address the failures in codecov/changes?

File: doc/en/example/markers.rst (hunk 1 above)
Comment (joshuastorck) [reply doc/en/example/markers.rst]: Removed

**Comment by joshuastorck** [comment]
Tests are passing. Pleas let me know if there are additional changes needed. Thank you!

**Comment by joshuastorck** [comment]
Thanks @nicoddemus. When should I expect this to land in a PyPI release?

**Comment by nicoddemus** [comment]
I think in the next couple of weeks we should have a 5.3 release. 👍

**Pull request merged**